src/components/Users/UserListAndCard.tsx

```diff
@@ -358,6 +359,13 @@ export default function UserListView({
   onTabChange,
 }: UserListViewProps) {
   const { t } = useTranslation();
+  const searchInputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (searchValue) {
+      searchInputRef.current?.focus();
+    }
+  }, [searchValue]);
```

Comment:
this hack is just hiding the root cause by solving the side effect by refocusing it back. why are we loosing the focus in the first place?

Reply:
because the `UserListAndCard.tsx` component re-renders every time the users prop changes, which occurs whenever the search value updates. This causes the focus in the SearchInput to be lost

Reply:
```
<UserListView
        users={userListData?.results ?? []}
        onSearch={(username) => updateQuery({ username })}
        searchValue={qParams.username}
        activeTab={activeTab}
        onTabChange={setActiveTab}
 />
  ```

Reply:
Move out the SearchInput back to the individual components that use UserListView, and that should take care of it.

Reply:
Got it

Reply:
Hey @Jacobjeevan, wait… this component is also used to view FacilityBasedUsers? 🤔

Reply:
Hey @Jacobjeevan, there’s nothing wrong with `UserListView` itself, as it’s working fine when used to view `FacilityUsers`. The issue seems to be with the `ManageUsers` component, which uses `UserListView`. It’s causing unnecessary re-renders whenever a user inputs anything.

Reply:
> Hey @Jacobjeevan, there’s nothing wrong with `UserListView` itself, as it’s working fine when used to view `FacilityUsers`. The issue seems to be with the `ManageUsers` component, which uses `UserListView`. It’s causing unnecessary re-renders whenever a user inputs anything.

Yes, hence move out the SearchInput into the components that use UserListView (in this case, both ManageUsers and FacilityUsers).

Reply:
`UserListView` is alright. Something to do with `ManageUsers` comp causing re-render the whole page as `UserListView` is working absolutely fine with `FacilityUsers`

Reply:
I just fixed it! 🥳

Reply:
> `UserListView` is alright. Something to do with `ManageUsers` comp causing re-render the whole page as `UserListView` is working absolutely fine with `FacilityUsers`

Remove the Loading checks that return <Loading /> in ManageUsers (and adjust the other variables accordingly), that should fix it.

Reply:
yup